File: 0002_codes/C - Kernel_Level/OS01.c
#include <stdio.h>
#include <stdlib.h>

/*
 * OS01 : environment variable explorer
 *
 * This program:
 * - prints the value of a specific environment variable
 * - shows how the operating system provides configuration
 * - lets you check variables like PATH, HOME, USER, USERNAME
 */

int main(void) {
    char variable_name[64];
    const char *value;

    printf("=========================================\n");
    printf("            OS01 - environment           \n");
    printf("=========================================\n");

    /* ask user for the name of the variable */
    printf("enter environment variable name: ");
    if (scanf("%63s", variable_name) != 1) {
        printf("invalid input\n");
        return 1;
    }

    /* read the variable */
    value = getenv(variable_name);

    if (value == NULL) {
        printf("variable '%s' not found\n", variable_name);
    } else {
        printf("%s = %s\n", variable_name, value);
    }

    printf("=========================================\n");

    return 0;
}

File: 0002_codes/JAVA - User_Level/OS01.java
import java.util.Scanner;

/*
 * OS01 - environment variable explorer
 * this program:
 * - asks for the name of an environment variable
 * - reads its value from the operating system
 * - prints the value or a message if it does not exist
 */

public class OS01 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("=========================================");
        System.out.println("            OS01 - environment           ");
        System.out.println("=========================================");
        System.out.print("enter environment variable name: ");

        String varName = scanner.nextLine().trim();

        if (varName.isEmpty()) {
            System.out.println("invalid input");
            System.out.println("=========================================");
            return;
        }

        /* read environment variable */
        String value = System.getenv(varName);

        if (value == null) {
            System.out.println("variable '" + varName + "' not found");
        } else {
            System.out.println(varName + " = " + value);
        }

        System.out.println("=========================================");
    }
}
